plugins/wallet_plugin/wallet.cpp

```diff
@@ -327,6 +327,16 @@ void wallet_api::unlock(string password)
 } EOS_RETHROW_EXCEPTIONS(chain::wallet_invalid_password_exception,
                           "Invalid password for wallet: \"${wallet_name}\"", ("wallet_name", get_wallet_filename())) }
 
+void wallet_api::check_password(string password)
+{ try {
+   FC_ASSERT(password.size() > 0);
+   auto pw = fc::sha512::hash(password.c_str(), password.size());
+   vector<char> decrypted = fc::aes_decrypt(pw, my->_wallet.cipher_keys);
+   auto pk = fc::raw::unpack<plain_keys>(decrypted);
+   FC_ASSERT(pk.checksum == pw);
+} EOS_RETHROW_EXCEPTIONS(chain::wallet_invalid_password_exception,
+                          "Invalid password for wallet: \"${wallet_name}\"", ("wallet_name", get_wallet_filename())) }
+
```

Comment:
This does exactly what unlock does minus the assignment which would not hurt. Seems like you could just call `unlock`

Reply:
I thought about that too but my concern is if I call `unlock()` on an already unlocked wallet. That will replace `_keys` with the decrypted keys. What if `_keys` has new keys that have not been written out to `cipher_keys` yet? Seems like I could potentially lose unsaved keys that way. I was a little surprised that `import_key()` for example doesn't save the wallet -- so `_keys` in that case will have contents that don't exist in `chiper_keys` until `encrypt_keys()` is called.

Basically I couldn't convince myself of the safety here.

Reply:
Maybe one approach to reducing code duplication here is to simply have unlock() call the new check_password()

Reply:
I'm a little nervous to mess with it...

Reply:
Actually now that I look at it again I see that we do save right after import/create. Still, I'm a little nervous touching this further at this juncture